lib/kvdb/ikvdb.c

```diff
@@ -394,15 +424,40 @@ ikvdb_storage_add(const char *kvdb_home, struct kvdb_cparams *params)
         return err;
     }
 
+    err = ikvdb_is_pmem_only(kvdb_home, &meta, &pmem_only);
+    if (err)
+        return err;
+
+    if (!pmem_only && (params->storage.mclass[MP_MED_CAPACITY].path[0] != '\0')) {
+        log_err("Cannot add %s to a standard KVDB %s: capacity mclass path must be provided "
+                "at create time", params->storage.mclass[MP_MED_CAPACITY].path, kvdb_home);
+        return merr(EINVAL);
+    }
+
+    if (pmem_only && (params->storage.mclass[MP_MED_STAGING].path[0] != '\0') &&
```

Comment:
So if you had a pmem-only, then you can't add a staging unless you have a capacity?

Reply:
Apart from pmem-only, I believe no other media class can be created without capacity. And as we don't support media class policy fallback, are staging-only & pmem-staging policies not valid?

Reply:
Yes, capacity is mandatory for a standard KVDB. So, when converting a pmem-only KVDB to a standard KVDB, either a capacity mclass must be added first or both capacity and storage mclass must be added in the same "hse storage add" command line...

The "staging-only" and "pmem-staging" policies determines only the data placement, i.e., where cN places the key/value blocks. So, these two are valid media class policies but they are not valid media class configurations for a KVDB.